FORMS.md

```diff
@@ -0,0 +1,192 @@
+# Creating and Using Forms
+
+This document lists specific guidelines for using our Form component and general forms guidelines.
+
+## General Form UI/UX
+
+### Labels, Placeholders, & Hints
+
+Labels are required for each input and should clearly mark the field. Optional text may appear below a field when a hint, suggestion, or context feels necessary. If validation fails on such a field, its error should clearly explain why without relying on the hint. Inline errors should always replace the microcopy hints. Placeholders should not be used as it’s customary for labels to appear inside form fields and animate them above the field when focused.
+
+Labels and hints are enabled by passing the appropriate props to each input:
+
+```
+<TextInput
+    label="Routing number"
+    hint="This is the 8 digit number in your bank check"
+/>
+```
+
+### Character Limits
+
+If a field has a character limit we should give that field a max limit and let the user know how many characters there are left outside of the input and below it.
+
+```
+<TextInput
+    maxLength={4}
+/>
+```
+
+### Native Keyboards
+
+We should always set people up for success on native platforms by enabling the best keyboard for the type of input we’re asking them to provide. See [keyboardType](https://reactnative.dev/docs/0.64/textinput#keyboardtype) in the React Native documentation.
+
+We have a couple of keyboard types [defined](https://github.com/Expensify/App/blob/572caa9e7cf32a2d64fe0e93d171bb05a1dfb217/src/CONST.js#L357-L360) and should be used like so:
+
+```
+<TextInput
+    keyboardType={CONST.KEYBOARD_TYPE.NUMBER_PAD}
+/>
+```
+
+### Autofill Behavior
+
+Forms should autofill information whenever possible i.e. they should work with browsers and password managers auto complete features.
+
+As a best practice we should avoid asking for information we can get via other means e.g. asking for City, State, and Zip if we can use Google Places to gather information with the least amount of hassle to the user.
+
+Browsers use the name parameter to autofill information into the input. Here's a [reference](https://developers.google.com/web/fundamentals/design-and-ux/input/forms#recommended_input_name_and_autocomplete_attribute_values) for names used by Chrome.
+
+```
+<TextInput
+    name="fname"
```

Comment:
What's the significance of "fname" here? 😕

Reply:
Oh, can we name it `firstName` then?

Reply:
Hmm I don't think so since the `name attribute` used by the browser for `given-name (first name)` is `fname`?

![Screen Shot 2022-03-02 at 1 01 25 PM](https://user-images.githubusercontent.com/22219519/156440366-514685e2-e15c-41a1-8249-16abc9993c43.png)

Reply:
Ah! Oh! Wow! Thanks for the clarification, after visiting the link this definitely makes sense 👍